src/privatesend/privatesend-client.cpp

```diff
@@ -268,13 +268,13 @@ void CPrivateSendClientSession::UnlockCoins()
     if (!privateSendClient.fEnablePrivateSend) return;
 
     while (true) {
-        TRY_LOCK(pwalletMain->cs_wallet, lockWallet);
+        TRY_LOCK(vpwallets[0]->cs_wallet, lockWallet);
```

Comment:
Not sure if this should be in the scope of this PR, but the more correct solution would be to have a pointer to `CWallet` as a member of `CPrivateSendClientSession` and use it everywhere. This member would be initialized once on startup so that it is set to `vpwallets[0]`. This would allow us to run private send multiple times in parallel in the future, for each wallet that the user has configured.

Reply:
I'd prefer that being out of scope :laughing: